### DISCUSSION TITLE
Support aarch64

### DISCUSSION THREAD

**Discussion by kelvinfan001** [vyperlang/vyper#2675] [Q&A] [answered]
Are there official releases of aarch64 binaries for the compiler? For example, when using [hardhat-vyper](https://hardhat.org/plugins/nomiclabs-hardhat-vyper.html), it downloads the x86 binary, causing it to be unusable for aarch64 machines.

**Comment by charles-cooper** [comment, marked as answer]
not yet. we could add it here https://github.com/vyperlang/vyper/blob/master/.github/workflows/build.yml. if you want to open a PR for this as well -- your contribution would be more than welcome!

**Reply by charles-cooper** [reply]
might be a one-liner once https://github.com/actions/virtual-environments/issues/4133 is fixed - just add `--target-architecture=universal2` in the `make freeze` rule for macos.

**Reply by kelvinfan001** [reply]
Thanks for your reply! Like this https://github.com/vyperlang/vyper/pull/2687?

**Reply by charles-cooper** [reply]
ah i think something more like this https://github.com/charles-cooper/vyper/commit/49a1b37718267221f4ea85b999312902a7f0b4f1

**Reply by kelvinfan001** [reply]
Updated :)

**Comment by charles-cooper** [comment]
@kelvinfan001 a workaround here -- https://github.com/tintinweb/vscode-vyper/issues/18#issuecomment-1101627345 -- maybe you can create an alias `vyper="arch -x86_64 <path/to/vyper/binary"` so that hardhat will run vyper in rosetta mode.
